src/s3path.jl

```diff
@@ -173,7 +181,7 @@ function FilePathsBase.parents(fp::S3Path)
     end
 end
 
-FilePathsBase.exists(fp::S3Path) = s3_exists(fp.config, fp.bucket, fp.key)
+FilePathsBase.exists(fp::S3Path) = s3_exists(fp.config, fp.bucket, fp.key; version=fp.version)
```

Comment:
Interesting tidbit I just discovered: this can be `s3_exists(fp.config, fp.bucket, fp.key; fp.version)`
```julia
julia> struct Demo
           x
           y
       end

julia> d = Demo(1,2)
Demo(1, 2)

julia> f(; y) = y
f (generic function with 1 method)

julia> f(; d.y)
2
```

Reply:
only in Julia 1.5+ unforunately, I love this syntax 😄

Reply:
FYI: https://github.com/JuliaLang/Compat.jl/pull/747. I need this for something else